src/runtime/thread_pool_common.h

```diff
@@ -52,6 +52,10 @@ struct work_queue_t {
     // whether the thread pool has been initialized.
     bool shutdown, initialized;
 
+    work_queue_t() {
+        halide_mutex_init(&mutex);
```

Comment:
This will cause double initialization when parallel is used on a pipeline for HVX. Firstly, the constructor will initialize the mutex here. Then based on the work queue having been initialized or not, halide_do_par_for in qurt_thread_pool.cpp will initialize it again (https://github.com/vgundlur/Halide/blob/hex-dma-qurt-mutex-init/src/runtime/qurt_thread_pool.cpp#L129).

Also, we use a custom loader on 8996 and unfortunately, that custom loader doesn't do well with constructors (it might still work, though) See this code (https://github.com/vgundlur/Halide/blob/hex-dma-qurt-mutex-init/src/runtime/qurt_init_fini.cpp) and this (https://github.com/vgundlur/Halide/blob/hex-dma-qurt-mutex-init/src/runtime/hexagon_remote/dlib.cpp#L439)